pkg/helpers/helpers.go

```diff
@@ -62,6 +85,30 @@ func DeleteCluster(ctx context.Context, id string, deprovision bool, connection
 	}
 }
 
+func ArchiveSubscription(ctx context.Context, id string, connection *sdk.Connection) {
+	connection.Logger().Info(ctx, "Archiving subscription '%s'", id)
+	// Send the request to patch subscription to archive
+	body := `{ "status": "Archived" }`
```

Comment:
This is a "soft delete" is it? So subsequent endpoint calls will just exclude these subscriptions? If that is the case, will we get the true endpoint performance since we will perhaps still execute some logic to skip archived clusters because we didn't hard delete?

Thoughts @chentex @dry923 ?

Reply:
So the end-point it's already doing this logic of skipping `soft deleted` subscriptions, so in my opinion it will be the real performance of it.
If we did hit a performance issue here, it would mean the `DB` is missing an index in that particular field, and would be a good feedback for the team. (I would highly doubt that is missing)

Reply:
Also I am still waiting on the review from Zohar, he might have more comments to do on this, since I asked him specifically to review this part.

Reply:
Sounds good!

Reply:
this should be DELETE on `/subscriptions/{id}`, which flags the Subscription as Deprovisioned and releases its resources.

Reply:
@chentex   if you're already deleting the cluster through CS, you shouldn't have to DELETE a subscription in AMS.  CS does that for you.